generators/gulp/templates/gulp/config.js

```diff
@@ -1,22 +1,24 @@
+<% isCraft = platform == 'craft' -%>
+<% var rootSrcPath = isCraft ? 'public' : 'dist' -%>
 //
 //   Config
 //
 //////////////////////////////////////////////////////////////////////
 
 var paths = {
-  dist: 'dist/',
+  dist: '<%= rootSrcPath %>/',
```

Comment:
Yep, you're right and I feel silly for not thinking of this. As much as I dislike the idea of adding a `dist` directing to `/public` (not even sure why it bothers me) this makes sense.